### PR TITLE
optimize: support where method condition(find_in_set)

### LINKED ISSUE #4565: mysql find_in_set不支持？
seata: 1.4.2
mysql find_in_set 不支持吗？
https://github.com/seata/seata/blob/76b3e88dde5cedec7ff53dfe40e81917be37307c/sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java#L72  我看这里似乎确实是不支持的

我有一个`update tablename set column = 1 where FIND_IN_SET(?, parent_ids)`,类似这样的语句，执行后报错了
```
execute executeAutoCommitTrue error:unexpected WHERE expr: SQLMethodInvokeExpr method: .s.r.d.exec.AbstractDMLBaseExecutor
java.lang.IllegalArgumentException: unexpected WHERE expr: SQLMethodInvokeExpr
at io.seata.sqlparser.druid.BaseRecognizer.executeVisit(BaseRecognizer.java:69)
at io.seata.sqlparser.druid.mysql.BaseMySQLRecognizer.getWhereCondition(BaseMySQLRecognizer.java:85)
```

### PR DESCRIPTION
**Pull request by doubleDimple**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
此pr用来支持where 条件后带函数导致报错,

### Ⅱ. Does this pull request fix one issue?

fixes #4565

### CODE REVIEW

**Review by caohdgege — APPROVED** [review]
LGTM，补充一下md文件

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * author: doubleDimple lovele.cn@gmail.com
+ */
+public class SupportSqlWhereMethod {
+
+    public static final List<String> SUPPORT_SQL_METHODS = new ArrayList<>();
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:26]: new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
且最好为单例,不要public直接给出去,而是给一个单例instance出去,再通过掉单例的add接口去添加支持的method

Comment (doubleDimple) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java]: get

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+public class SupportSqlWhereMethod {
+
+    private static final Set<String> SUPPORT_SQL_METHODS = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+    private static class SupportSqlWhereMethodHolder {
+        private static final SupportSqlWhereMethod INSTANCE = new SupportSqlWhereMethod();
+    }
+
+    public static SupportSqlWhereMethod getInstance() {
+        add();
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:33]: 这不太对呀,FIND_IN_SET写在staic{}中,下面的add做成public ,其它地方可以掉SupportSqlWhereMethod.getInstance).add(method)

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * author: doubleDimple lovele.cn@gmail.com
+ */
+public class SupportSqlWhereMethod {
+
+    private static final Set<String> SUPPORT_SQL_METHODS = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:26]: static 去掉,通过调用getInstance拿到实例后进行add添加

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    private static class SupportSqlWhereMethodHolder {
+        private static final SupportSqlWhereMethod INSTANCE = new SupportSqlWhereMethod();
+    }
+
+    public static SupportSqlWhereMethod getInstance() {
+        add();
+        return SupportSqlWhereMethodHolder.INSTANCE;
+    }
+
+    private static void add() {
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:37]: public void add(String method)

**Review by a364176773 — CHANGES_REQUESTED** [review]

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java (hunk 2 above)
Comment (doubleDimple) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:33]: get

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java (hunk 3 above)
Comment (doubleDimple) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:26]: get

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java (hunk 4 above)
Comment (doubleDimple) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:37]: get

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    /**
+     * 
+     * @param methodName
+     * @return boolean
+     */
+    public boolean checkIsSupport(String methodName) {
+
+        if (supportMethodNames.contains(methodName)) {
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:51]: return supportMethodNames.contains(methodName)

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+public class SupportSqlWhereMethod {
+
+    private final Set<String> supportMethodNames = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+    private static class SupportSqlWhereMethodHolder {
+        private static final SupportSqlWhereMethod INSTANCE = new SupportSqlWhereMethod();
+    }
+
+    static {
+        getInstance().add("FIND_IN_SET");
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:33]: 写在SupportSqlWhereMethod 构造函数里比较好 public SupportSqlWhereMethod(){
add("FIND_IN_SET");
}

Comment (doubleDimple) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:33]: get

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java (hunk 5 above)
Comment (doubleDimple) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/SupportSqlWhereMethod.java:51]: get

**Comment by pengten** [comment]
好多非必要的代码格式化 orz

File: changes/en-us/2.0.0.md [hunk 7]
```diff
@@ -7,31 +7,32 @@
...
+- [[#1234](https://github.com/seata/seata/pull/1234)] Please delete the sample later
 
-   ### optimize：
-  - [[#1234](https://github.com/seata/seata/pull/1234)] Please delete the sample later
+### optimize：
 
+- [[#4567](https://github.com/seata/seata/pull/4567)] support where method condition(find_in_set)
 
-  ### test:	
+### test:
```
Comment (pengten) [inline changes/en-us/2.0.0.md:29]: 好多非必要的代码格式化 orz

**Review by a364176773 — APPROVED** [review]
LGTM

**Pull request merged**